pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -224,14 +233,27 @@ func readEntity(rows *sql.Rows, r FieldSelectRequest) (*entity.Entity, error) {
 }
 
 func (s *sqlEntityServer) Read(ctx context.Context, r *entity.ReadEntityRequest) (*entity.Entity, error) {
+	ctx, span := s.tracer.Start(ctx, "storage_server.Read")
```

Comment:
this is up to you, but since `sqlEntityServer` implements the `entity.EntityStoreServer` interface, one option that might work is to create wrappers for tracing and error logs so that `sqlEntityServer` or future implementations of `entity.EntityStoreServer` don't need to implement them directly.

```go
type EntityServerTracer struct {
  original entity.EntityStoreServer
}

func (s *EntityServerTracer) Read(ctx context.Context, r *entity.ReadEntityRequest) (*entity.Entity, error) {
	ctx, span := s.tracer.Start(ctx, "storage_server.Read")
	defer span.End()
	span.SetAttributes(attribute.String("storage_server.request", r.String()))
	return s.original.Read(ctx, r)
}
```

the same thing could be done for intercepting the errors and logging them with the method name

Reply:
If we did this and implemented traces in the wrapper, this looks like we'd only get a single span for each of these gRPC endpoints/functions. I'd like to have traces for a lot of the internal calls these made as well, since at this stage was don't really know where bottlenecks will be.